libnetwork/sandbox_externalkey_unix.go

```diff
@@ -24,82 +25,68 @@ const (
 	success         = "success"
 )
 
+func init() {
+	// TODO(thaJeztah): should this actually be registered on FreeBSD, or only on Linux?
+	reexec.Register("libnetwork-setkey", processSetKeyReexec)
+}
+
+type setKeyData struct {
+	ContainerID string
+	Key         string
+}
+
 // processSetKeyReexec is a private function that must be called only on an reexec path
 // It expects 3 args { [0] = "libnetwork-setkey", [1] = <container-id>, [2] = <short-controller-id> }
 // It also expects specs.State as a json string in <stdin>
 // Refer to https://github.com/opencontainers/runc/pull/160/ for more information
 // The docker exec-root can be specified as "-exec-root" flag. The default value is "/run/docker".
 func processSetKeyReexec() {
-	var err error
-
-	// Return a failure to the calling process via ExitCode
-	defer func() {
-		if err != nil {
-			logrus.Fatalf("%v", err)
-		}
-	}()
+	if err := setKey(); err != nil {
+		println(err)
```

Comment:
The bootstrapping `println` builtin [is not technically part of the Go language.](https://go.dev/ref/spec#Bootstrapping)
```suggestion
		fmt.Println(err)
```

Reply:
Oh! Good call yes; that was originally a "WIP" commit, and I must've been lazy when I wrote that code (`println` prints to stderr, so saved me from having to use `os.StdOut` etc).

Let me update